src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -2530,10 +2530,47 @@ protected function get_retry_interval() {
 		 *
 		 * @since TBD
 		 *
-		 * @param int An interval in seconds; defaults to the record frequency / 2.
+		 * @param int $retry_interval An interval in seconds; defaults to the record frequency / 2.
 		 * @param Tribe__Events__Aggregator__Record__Abstract $this
 		 */
 		return apply_filters( 'tribe_aggregator_scheduled_records_retry_interval', $retry_interval, $this );
 	}
+
+	/**
+	 * Returns the record retry timestamp.
+	 *
+	 * @since TBD
+	 *
+	 * @return int|bool Either the record retry timestamp or `false` if the record will
+	 *                  not retry to import.
+	 */
+	public function get_retry_time() {
+		$retry_interval = $this->get_retry_interval();
+
+		if ( empty( $retry_interval ) ) {
+			return false;
+		}
+
+		if ( ! $this->get_last_import_status( 'error', true ) ) {
+			return false;
+		}
+
+		$last_attempt_time = strtotime( $this->last_child()->post->post_modified_gmt );
```

Comment:
Should we do some validation on the date? It might be invalid.

Reply:
The code is this:
```php
$last_attempt_time = strtotime( $this->last_child()->post->post_modified_gmt );
$retry_time = $last_attempt_time + (int) $retry_interval;

if ( $retry_time < time() ) {
    $retry_time = false;
}
```

So if the `strtotime` function cannot parse the `post_modified_gmt` it will return `false`.
Then `retry_time` will be `0 + something`.
Since we're not at the beginning of times `retry_time` will be `< time()` and `retry_time` will be then set to `false`

Reply:
perfecto!